crates/common/tedge_config_macros/impl/src/reader.rs

```diff
@@ -334,14 +340,50 @@ fn reader_value_for_field<'a>(
                 }
                 FieldDefault::FromKey(default_key) | FieldDefault::FromOptionalKey(default_key) => {
                     observed_keys.push(default_key);
+                    // TODO error if the field touches an unrelated multi?
```

Comment:
Oops, this is done now, and tested, although the error message https://github.com/thin-edge/thin-edge.io/pull/3126/commits/85b30f82e42672999fdaa6389f2bccbd882a8bba#diff-407a6572ceebfdf137f7e0369291214688fa368a92f6381eb18689aa5d8eb982R374  isn't very informative